### PR TITLE
[#202] Added Picker button

### LINKED ISSUE #202: PickerButton
Screen Shot 2021-06-29 at 10 08 44 AM

### PR DESCRIPTION
**Pull request by alexandraB99**
Adding picker button
* As an example it opens Qt's ColorDialog, as soon as
  a color is selected there, the button is colored
  with that. Final color picker to be implemented in
  a seperate task.

Also minor improvements in main.qml and sandbox.pro

fixes #202

### CODE REVIEW

**Review by PascalPrecht — CHANGES_REQUESTED** [review]
Nice PR @alexandraB99 !

I've tested this locally and it seems to work great! There's a few things I've noticed that need adjustment though. I've left some comments inline. Let me know what you think.

I've also noticed that the color selector renders all black when it opens. Not sure if that is something that's in our control.
Screenshot 2021-07-08 at 16 57 18

Bonus points if you update your commit message to look more like something like:

```
feat(Controls): introduce StatusColorSelectorButton

# your message body

Closes #202
```

Where `StatusColorSelectorButton` is just another name I've proposed as `PickerButton` doesn't really tell us anything.

Lemme know if I can help with anything!

File: sandbox/Buttons.qml [hunk 1]
```diff
@@ -1,239 +1,261 @@
...
+        onClicked: {
+            colorDialog.open();
+        }
+        ColorDialog {
+            id: colorDialog
+            property bool colorSelected: false
+            onAccepted: {
+                colorSelected = true;
+            }
+        }
```
[the remark this answers is not in the document]
Comment (PascalPrecht) [reply sandbox/Buttons.qml:259]: As discussed offline, I totally confused how this thing is behaving. There's no color selector shipping with that button.

Ignore all the comments made above! 🙏

File: src/StatusQ/Controls/qmldir [hunk 2]
```diff
@@ -15,3 +15,4 @@ StatusRadioButton 0.1 StatusRadioButton.qml
 StatusCheckBox 0.1 StatusCheckBox.qml
 StatusSlider 0.1 StatusSlider.qml
 StatusBaseInput 0.1 StatusBaseInput.qml
+StatusPickerButton 0.1 StatusPickerButton.qml
```
[the remark this answers is not in the document]
Comment (PascalPrecht) [reply src/StatusQ/Controls/qmldir:18]: As mentioned here https://github.com/status-im/StatusQ/pull/247#discussion_r666294544 I totally confused the semantics of this component. No rename needed!

**Comment by PascalPrecht** [comment]
Once https://github.com/status-im/StatusQ/pull/247#discussion_r666268636 and https://github.com/status-im/StatusQ/pull/247#discussion_r666271034 are addressed we can merge this!

File: src/StatusQ/Controls/StatusPickerButton.qml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        Label {
+            id: textLabel
+            anchors.fill: parent
+            anchors.margins: 5
+            verticalAlignment: Text.AlignVCenter
+            font.pixelSize: 15
+            font.family: Theme.palette.baseFont.name
+            color: root.contentColor
+            text: root.text
+        }
```
[the remark this answers is not in the document]
Comment (alexandraB99) [reply src/StatusQ/Controls/StatusPickerButton.qml:44]: Left and right margins added :) Also Label replaced with a StatusBaseText. As about the color, I see in Figma the text has #939BA1 which points to baseColor1, what am I missing?

[the remark this answers is not in the document]
Comment (PascalPrecht) [reply src/StatusQ/Controls/StatusPickerButton.qml:47]: Looks like we're looking at different designs! :D

[the remark this answers is not in the document]
Comment (PascalPrecht) [reply src/StatusQ/Controls/StatusPickerButton.qml:47]: So to clarify, the color change needed was in the icon (see: https://github.com/status-im/StatusQ/pull/247#discussion_r666271034) not the text

**Review by PascalPrecht — APPROVED** [review]

**Comment by PascalPrecht** [comment]
Landed 🎉 awesome work!

**Pull request merged**